File: Algorithms/487. Max Consecutive Ones II/golang/solution.sliding-window.go
func findMaxConsecutiveOnes(nums []int) int {
	var longestSequence, left, right, numZeroes int = 0, 0, 0, 0

	for right < len(nums) {
		if nums[right] == 0 {
			numZeroes++
		}

		for numZeroes == 2 {
			if nums[left] == 0 {
				numZeroes--
			}
			left++
		}

		if right-left+1 > longestSequence {
			longestSequence = right - left + 1
		}
		right++
	}

	return longestSequence
}

File: Algorithms/487. Max Consecutive Ones II/kotlin/solution.sliding-window.kt
class Solution {
    fun findMaxConsecutiveOnes(nums: IntArray): Int {
        var longestSequence: Int = 0
        var left: Int = 0
        var right: Int = 0
        var numZeroes: Int = 0

        while (right < nums.size) {
            if (nums[right] == 0) {
                numZeroes++
            }

            while (numZeroes == 2) {
                if (nums[left] == 0) {
                    numZeroes--
                }
                left++
            }

            longestSequence = maxOf(longestSequence, right - left + 1)
            right++
        }

        return longestSequence
    }
}
